Public/Sdk/SelfHost/BuildXL/Tools/QTest/Tool.QTestRunner.dsc

```diff
@@ -213,6 +214,37 @@ export function runQTest(args: QTestArguments): Result {
         )
     );
     const qTestLogsDir: StaticDirectory = result.getOutputDirectory(logDir);
+
+    // If code coverage is enabled, schedule a pip that will perform coverage file upload.
+    if (qCodeCoverageEnumType === "DynamicCodeCov"){
+        let coverageLogDir = args.coverageLogsDir || Context.getNewOutputDirectory("coverageLogs");
+        let coverageConsolePath = p`${coverageLogDir}/coverageUpload.stdout`;
+
+        let commandLineArgsForUploadPip: Argument[] = [
+            Cmd.option("--qTestLogsDir ", Artifact.output(coverageLogDir)),
+            Cmd.option("--qTestContextInfo ", Artifact.input(qTestContextInfoPath)),
+            Cmd.option("--coverageDirectory ", Artifact.input(qTestLogsDir)),
+            Cmd.option("--qTestBuildType ", args.qTestBuildType),
+            Cmd.option("--qtestPlatform ", qTestPlatformToString(args.qTestPlatform))
+        ];
+
+        Transformer.execute(
+            Object.merge<Transformer.ExecuteArguments>(
+                {
+                    tool: args.qTestTool ? args.qTestTool : qTestTool,
+                    tags: tags,
+                    description: "QTest Coverage Upload",
+                    arguments: commandLineArgsForUploadPip,
+                    consoleOutput: coverageConsolePath,
+                    workingDirectory: tempDirectory,
+                    disableCacheLookup: true,
+                    privilegeLevel: args.privilegeLevel,
+                },
+                untrackingCBPaths

```

Comment:
This is a weird and unnecessary use of `Object.merge`. 

Instead of defining `untrackingCBPaths` as
```
        untrackingCBPaths =  {
            unsafe: {
                untrackedPaths: [
                    qTestContextInfoFile,
                ],
                untrackedScopes: [
                    d`d:/data`,
                    d`d:/app`,
                    ...addIf(Environment.hasVariable("QAUTHMATERIALROOT"), Environment.getDirectoryValue("QAUTHMATERIALROOT")),
                ]
            }
        };
```
you should define
```
        unsafeOptions = {
            untrackedPaths: [
                qTestContextInfoFile,
            ],
            untrackedScopes: [
                d`d:/data`,
                d`d:/app`,
                ...addIf(Environment.hasVariable("QAUTHMATERIALROOT"), Environment.getDirectoryValue("QAUTHMATERIALROOT")),
            ]
        };
```
so that here you can simply say
```
Transformer.execute({
   ... // 
   unsafe: unsafeOptions
});
```

Reply:
An aside: up above there is a line like this
```typescript
let tags = Object.merge<string[]>(args.tags, defaultArgs.tags);
```
There are a couple of idiomatic ways to concatenate two arrays e.g., 
```typescript
let result = [...arr1, ...arr2];
```
or
```typescript
let result = arr1.concat(arr2);`
```
`Object.merge` is not one of them.

Reply:
Thanks for fixing this